frontend/src/pages/retailer/ModalAddRetailer.js

```diff
@@ -0,0 +1,264 @@
+                });
+
+    };
+
+    const handleCheckedChange = (warehouseId) => {
+            checkBoxes.push(warehouseId);
+    };
+
+
+
+    const closeModalEdit = (e) => {
+        setEditWarehouse(
+            preState => ({
+                ...preState,
+                editShow: false
+            }));
+
+    };
+
+    const closeModalAdd = (e) => {
+        setLgShow(e);
+    };
+
+    function deleteWarehouse() {
+        const filtered = warehouses.filter(
+            function (e) {
+                return this.indexOf(e) < 0;
+            },
+            checkBoxes
+        );
+        setWarehouses(filtered);
+    }
+
+    const tableRows = warehouses.filter(warehouse => warehouse != null).map(warehouse => (
+        <tr key={warehouse.identifier}>
+            <td>{warehouse.identifier}</td>
+            <td>{warehouse.type}</td>
+            <td>{warehouse.addressDto.city}, {warehouse.addressDto.addressLine1},
+                {warehouse.addressDto.addressLine2}, {warehouse.addressDto.state.state}</td>
+            <td>{warehouse.totalCapacity}</td>
+            <td><FaEdit style={{textAlign: 'center', color: '#1a7fa8'}}
+                        size={'1.3em'}
+                        onClick={() => {
+                            setEditWarehouse({
+                                editShow: true,
+                                warehouse: warehouse
+                            });
+                        }}/>
+            </td>
+            <td>
+                <input type="checkbox" onClick={() => handleCheckedChange(warehouse)}/>
+            </td>
+        </tr>
+    ));
+
+    const modals =
+        <React.Fragment>
+
+            <ModalEditWarehouseRetailer props={editWarehouse} onEditWarehouse={editWarehouseHandler} onChange={closeModalEdit}
+                                        currentCustomerId={currentCustomerId}/>
+            <ModalAddWarehouseRetailer props={lgShow} onAddWarehouse={addWarehouseHandler} onChange={closeModalAdd}
+                                       currentCustomerId={currentCustomerId}
+            />
+        </React.Fragment>;
+
+    const body =
+        <React.Fragment>
+            <Table hover size="sm">
+                <thead>
+                <tr>
+                    <th>Identifier</th>
+                    <th>Type</th>
+                    <th>Full Address</th>
+                    <th>Total Capacity</th>
+                </tr>
+                </thead>
+                <tbody>
+                {tableRows}
+                </tbody>
+            </Table>
+        </React.Fragment>;
+
+    const header =
+        <React.Fragment>
+            <Row>
+                <Col>
+                    <Button className="mainButton" size="sm" onClick={() => setLgShow(true)}>
+                        Add
+                    </Button>
+
+                </Col>
+                <Col md={10}>
+                    <Button className="mainButton" size="sm" onClick={() => deleteWarehouse()}>
```

Comment:
We may pass a function reference instead of anonymous function onClick={deleteWarehouse}